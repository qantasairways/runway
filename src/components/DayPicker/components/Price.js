import React from 'react';
import PropTypes from 'prop-types';
import { keyframes } from 'emotion';

import { colours, mq, fontWeight } from '../../../theme/airways';
import { CSS_PSEUDO_AFTER } from '../../../constants/css';
import { abbrNum } from '../helpers';

const priceFontStyles = {
  fontSize: '0.8125rem',
  fontWeight: fontWeight.bold,
  lineHeight: 1,
  [mq.medium]: {
    fontSize: '0.875rem'
  }
};

const loadingPrice = keyframes`
  from {
    background-position: -26px 0px
  }
  to {
    background-position:0
  }
`;

function Price({
  isLoadingPrice,
  value,
  taxValue,
  currencySymbol,
  isLowestPrice,
  points,
  isClassic,
  priceInPoints
}) {
  return (
    <div>
      {isLoadingPrice && (
        <div
          css={{
            ...priceFontStyles,
            position: 'absolute',
            left: '50%',
            width: '26px',
            marginLeft: '-13px',
            [CSS_PSEUDO_AFTER]: {
              content: '""',
              position: 'absolute',
              left: 0,
              right: 0,
              top: '2px',
              height: '7px',
              margin: '0 auto',
              animation: `${loadingPrice} 2s linear infinite`,
              background:
                'linear-gradient(to right,rgba(0,0,0,0.06) 20%,rgba(0,0,0,0.01) 70%,rgba(0,0,0,0.05) 100%)',
              bottom: '1px',
              boxShadow: '0 0 1px rgba(0,0,0,0.12)',
              borderRadius: '4px'
            },
            [mq.medium]: {
              width: '30px',
              marginLeft: '-15px',
              [CSS_PSEUDO_AFTER]: {
                height: '10px'
              }
            }
          }}
        />
      )}
      {!isLoadingPrice && (
        <div
          css={{
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'center',
            [mq.medium]: {
              flexDirection: 'row',
              alignItems: 'center'
            }
          }}
        >
          <div
            css={{
              color: isLowestPrice ? '#008600' : colours.mediumDarkGrey,
              ...priceFontStyles
            }}
          >
            {priceInPoints ? abbrNum(points) : currencySymbol + abbrNum(value)}
          </div>
          {taxValue &&
            (priceInPoints && isClassic ? (
              <div
                css={{
                  fontSize: '0.75rem',
                  lineHeight: 1,
                  color: colours.mediumDarkGrey,
                  fontWeight: fontWeight.bold,
                  [mq.medium]: {
                    fontSize: '0.875rem',
                    marginLeft: '2px'
                  }
                }}
              >
                <span
                  css={{
                    paddingRight: 0,
                    [mq.medium]: {
                      paddingRight: '5px'
                    }
                  }}
                >
                  +
                </span>
                <span>{currencySymbol}</span>
                <span>{abbrNum(taxValue)}</span>
                <span>^</span>
              </div>
            ) : null)}
        </div>
      )}
    </div>
  );
}

Price.propTypes = {
  isLoadingPrice: PropTypes.bool,
  value: PropTypes.number,
  taxValue: PropTypes.number,
  points: PropTypes.number,
  currencySymbol: PropTypes.string,
  isLowestPrice: PropTypes.bool,
  isClassic: PropTypes.bool,
  priceInPoints: PropTypes.bool
};

Price.defaultProps = {
  isLoadingPrice: false,
  value: null,
  taxValue: null,
  points: null,
  currencySymbol: '',
  isLowestPrice: false,
  isClassic: null,
  priceInPoints: false
};

export default Price;
