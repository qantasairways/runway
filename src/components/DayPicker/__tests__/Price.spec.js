import React from 'react';
import { shallow } from 'enzyme';
import Price from '../components/Price';

describe('Price', () => {
  let component;

  const props = {
    isLoadingPrice: false,
    value: 12345,
    taxValue: 345,
    currencySymbol: '$',
    isLowestPrice: false,
    isDesktopDevice: false
  };

  it('renders correctly with defaults', () => {
    component = shallow(<Price />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with props', () => {
    component = shallow(<Price {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly when isLoadingPrice is true', () => {
    component = shallow(<Price {...props} isLoadingPrice />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly when isLowestPrice is true', () => {
    component = shallow(<Price {...props} isLowestPrice />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly when isDesktopDevice is true', () => {
    component = shallow(<Price {...props} isDesktopDevice />);

    expect(component).toMatchSnapshot();
  });
});
