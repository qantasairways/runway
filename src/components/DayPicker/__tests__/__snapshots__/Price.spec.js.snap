// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price renders correctly when isDesktopDevice is true 1`] = `
.emotion-4 {
  margin-top: 3px;
  width: 100%;
}

@media only screen and (min-width:768px) {
  .emotion-4 {
    margin-top: 7px;
  }
}

.emotion-0 {
  font-size: 0.8125rem;
  font-weight: 500;
  line-height: 1;
  position: absolute;
  left: 50%;
  width: 26px;
  margin-left: -13px;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    width: 30px;
    margin-left: -15px;
  }

  .emotion-0::after {
    height: 10px;
  }
}

.emotion-0::after {
  content: "";
  position: absolute;
  left: 0;
  right: 0;
  top: 2px;
  height: 7px;
  margin: 0 auto;
  -webkit-animation: animation-h2aas6 2s linear infinite;
  animation: animation-h2aas6 2s linear infinite;
  background: linear-gradient(to right,rgba(0,0,0,0.06) 20%,rgba(0,0,0,0.01) 70%,rgba(0,0,0,0.05) 100%);
  bottom: 1px;
  box-shadow: 0 0 1px rgba(0,0,0,0.12);
  border-radius: 4px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

.emotion-1 {
  color: #888888;
  font-size: 0.8125rem;
  font-weight: 500;
  line-height: 1;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    font-size: 0.875rem;
  }
}

.emotion-2 {
  font-size: 0.75rem;
  line-height: 1;
  color: #888888;
}

@media only screen and (min-width:768px) {
  .emotion-2 {
    font-size: 0.8125rem;
    margin-left: 2px;
  }
}

<div
  className="emotion-4"
>
  <div
    className="emotion-0"
  />
  <div
    className="emotion-3"
  >
    <div
      className="emotion-1"
    >
      $
      12.4k
    </div>
    <div
      className="emotion-2"
    >
      <span>
        +
      </span>
      <span>
        $
      </span>
      <span>
        345
      </span>
      <span>
         ^
      </span>
    </div>
  </div>
</div>
`;

exports[`Price renders correctly when isLoadingPrice is true 1`] = `
.emotion-1 {
  margin-top: 3px;
  width: 100%;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    margin-top: 7px;
  }
}

.emotion-0 {
  font-size: 0.8125rem;
  font-weight: 500;
  line-height: 1;
  position: absolute;
  left: 50%;
  width: 26px;
  margin-left: -13px;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    width: 30px;
    margin-left: -15px;
  }

  .emotion-0::after {
    height: 10px;
  }
}

.emotion-0::after {
  content: "";
  position: absolute;
  left: 0;
  right: 0;
  top: 2px;
  height: 7px;
  margin: 0 auto;
  -webkit-animation: animation-h2aas6 2s linear infinite;
  animation: animation-h2aas6 2s linear infinite;
  background: linear-gradient(to right,rgba(0,0,0,0.06) 20%,rgba(0,0,0,0.01) 70%,rgba(0,0,0,0.05) 100%);
  bottom: 1px;
  box-shadow: 0 0 1px rgba(0,0,0,0.12);
  border-radius: 4px;
}

<div
  className="emotion-1"
>
  <div
    className="emotion-0"
  />
</div>
`;

exports[`Price renders correctly when isLowestPrice is true 1`] = `
.emotion-4 {
  margin-top: 3px;
  width: 100%;
}

@media only screen and (min-width:768px) {
  .emotion-4 {
    margin-top: 7px;
  }
}

.emotion-0 {
  font-size: 0.8125rem;
  font-weight: 500;
  line-height: 1;
  position: absolute;
  left: 50%;
  width: 26px;
  margin-left: -13px;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    width: 30px;
    margin-left: -15px;
  }

  .emotion-0::after {
    height: 10px;
  }
}

.emotion-0::after {
  content: "";
  position: absolute;
  left: 0;
  right: 0;
  top: 2px;
  height: 7px;
  margin: 0 auto;
  -webkit-animation: animation-h2aas6 2s linear infinite;
  animation: animation-h2aas6 2s linear infinite;
  background: linear-gradient(to right,rgba(0,0,0,0.06) 20%,rgba(0,0,0,0.01) 70%,rgba(0,0,0,0.05) 100%);
  bottom: 1px;
  box-shadow: 0 0 1px rgba(0,0,0,0.12);
  border-radius: 4px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

.emotion-2 {
  font-size: 0.75rem;
  line-height: 1;
  color: #888888;
}

@media only screen and (min-width:768px) {
  .emotion-2 {
    font-size: 0.8125rem;
    margin-left: 2px;
  }
}

.emotion-1 {
  color: #008600;
  font-size: 0.8125rem;
  font-weight: 500;
  line-height: 1;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    font-size: 0.875rem;
  }
}

<div
  className="emotion-4"
>
  <div
    className="emotion-0"
  />
  <div
    className="emotion-3"
  >
    <div
      className="emotion-1"
    >
      $
      12.4k
    </div>
    <div
      className="emotion-2"
    >
      <span>
        +
      </span>
      <span>
        $
      </span>
      <span>
        345
      </span>
      <span>
         ^
      </span>
    </div>
  </div>
</div>
`;

exports[`Price renders correctly with defaults 1`] = `
.emotion-3 {
  margin-top: 3px;
  width: 100%;
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    margin-top: 7px;
  }
}

.emotion-0 {
  font-size: 0.8125rem;
  font-weight: 500;
  line-height: 1;
  position: absolute;
  left: 50%;
  width: 26px;
  margin-left: -13px;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    width: 30px;
    margin-left: -15px;
  }

  .emotion-0::after {
    height: 10px;
  }
}

.emotion-0::after {
  content: "";
  position: absolute;
  left: 0;
  right: 0;
  top: 2px;
  height: 7px;
  margin: 0 auto;
  -webkit-animation: animation-h2aas6 2s linear infinite;
  animation: animation-h2aas6 2s linear infinite;
  background: linear-gradient(to right,rgba(0,0,0,0.06) 20%,rgba(0,0,0,0.01) 70%,rgba(0,0,0,0.05) 100%);
  bottom: 1px;
  box-shadow: 0 0 1px rgba(0,0,0,0.12);
  border-radius: 4px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media only screen and (min-width:768px) {
  .emotion-2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

.emotion-1 {
  color: #888888;
  font-size: 0.8125rem;
  font-weight: 500;
  line-height: 1;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    font-size: 0.875rem;
  }
}

<div
  className="emotion-3"
>
  <div
    className="emotion-0"
  />
  <div
    className="emotion-2"
  >
    <div
      className="emotion-1"
    >
      0
    </div>
  </div>
</div>
`;

exports[`Price renders correctly with props 1`] = `
.emotion-4 {
  margin-top: 3px;
  width: 100%;
}

@media only screen and (min-width:768px) {
  .emotion-4 {
    margin-top: 7px;
  }
}

.emotion-0 {
  font-size: 0.8125rem;
  font-weight: 500;
  line-height: 1;
  position: absolute;
  left: 50%;
  width: 26px;
  margin-left: -13px;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    width: 30px;
    margin-left: -15px;
  }

  .emotion-0::after {
    height: 10px;
  }
}

.emotion-0::after {
  content: "";
  position: absolute;
  left: 0;
  right: 0;
  top: 2px;
  height: 7px;
  margin: 0 auto;
  -webkit-animation: animation-h2aas6 2s linear infinite;
  animation: animation-h2aas6 2s linear infinite;
  background: linear-gradient(to right,rgba(0,0,0,0.06) 20%,rgba(0,0,0,0.01) 70%,rgba(0,0,0,0.05) 100%);
  bottom: 1px;
  box-shadow: 0 0 1px rgba(0,0,0,0.12);
  border-radius: 4px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

.emotion-1 {
  color: #888888;
  font-size: 0.8125rem;
  font-weight: 500;
  line-height: 1;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    font-size: 0.875rem;
  }
}

.emotion-2 {
  font-size: 0.75rem;
  line-height: 1;
  color: #888888;
}

@media only screen and (min-width:768px) {
  .emotion-2 {
    font-size: 0.8125rem;
    margin-left: 2px;
  }
}

<div
  className="emotion-4"
>
  <div
    className="emotion-0"
  />
  <div
    className="emotion-3"
  >
    <div
      className="emotion-1"
    >
      $
      12.4k
    </div>
    <div
      className="emotion-2"
    >
      <span>
        +
      </span>
      <span>
        $
      </span>
      <span>
        345
      </span>
      <span>
         ^
      </span>
    </div>
  </div>
</div>
`;
